src/HealthChecks.AzureDigitalTwin/AzureDigitalTwinHealthCheck.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Microsoft.Azure.Management.DigitalTwins;
+using Microsoft.Azure.Management.ResourceManager.Fluent;
+using Microsoft.Azure.Management.ResourceManager.Fluent.Authentication;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Rest;
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace HealthChecks.AzureDigitalTwin
+{
+    public class AzureDigitalTwinHealthCheck
+           : IHealthCheck
+    {
+        private static readonly ConcurrentDictionary<string, AzureDigitalTwinsManagementClient> _managementClientConnections
+            = new ConcurrentDictionary<string, AzureDigitalTwinsManagementClient>();
+
+        private readonly string _clientId;
+        private readonly string _clientSecret;
+        private readonly string _tenantId;
+        private readonly ServiceClientCredentials _credentials;
+
+        public AzureDigitalTwinHealthCheck(string clientId, string clientSecret, string tenantId)
+        {
+            _clientId = (!string.IsNullOrEmpty(clientId)) ? clientId : throw new ArgumentNullException(nameof(clientId));
+            _clientSecret = (!string.IsNullOrEmpty(clientSecret)) ? clientSecret : throw new ArgumentNullException(nameof(clientSecret));
+            _tenantId = (!string.IsNullOrEmpty(tenantId)) ? tenantId : throw new ArgumentNullException(nameof(tenantId));
+
+            _credentials = new AzureCredentialsFactory().FromServicePrincipal(_clientId, _clientSecret, _tenantId, AzureEnvironment.AzureGlobalCloud);
+        }
+
+        public AzureDigitalTwinHealthCheck(ServiceClientCredentials credentials)
+        {
+            _credentials = (credentials is not null) ? credentials : throw new ArgumentNullException(nameof(credentials));
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                if (!_managementClientConnections.TryGetValue(_clientId, out var managementClient))
```

Comment:
See https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/pull/847 . The same problem may ne here.